base/strings/basic.jl

```diff
@@ -39,6 +39,7 @@ getindex(s::AbstractString, v::AbstractVector{<:Integer}) =
 getindex(s::AbstractString, v::AbstractVector{Bool}) =
     throw(ArgumentError("logical indexing not supported for strings"))
 
+get(s::AbstractString, i::Integer, default) = start(s) <= i <= endof(s) ? s[i] : default
```

Comment:
Nice, I tried `checkbounds()` but that throws an error.  `isvalid()` works great.